feat(cloudflare): add astro assets image service (#34)
withastro/adapters: packages/cloudflare/README.md

  });
```

### `imageService`

> `imageService: 'external' | 'passthrough'`

Reviewer: ```suggestion
`imageService: "external" | "passthrough"`
default `"passthrough"`
```
Just keeping it consistent!

Author: The default value is not `passthrough`, it is a value the user shouldn't set, but the behaviour is in line with `passthrough`, so I'm unsure if we should put a default here.

Reviewer: So the default is whichever service the main config has set or defaulted to?

Author: The default is whatever the user has as configuration, and if it is unsupported (sharp/squoosh) it is same as `passthrough`

Reviewer: Ok, that makes sense. And this is more like an override of the user config.

Could the cloudflare image service be offered directly?
```ts
export default defineConfig({
  image: {
    service: '@astrojs/cloudflare/image-service',
  }
})
```
I'm guessing this would work already too.

Author: Yeah that code-snippet should work

Reviewer: I'm wondering if it's a good idea to offer two ways of doing the same thing, but I don't feel strongly either way. I'll leave that upto you.

Author: I'm coming from: https://docs.astro.build/en/guides/integrations-guide/vercel/#imageservice
To make it "similar" ish, and I think you can do it both ways for Vercel too. But not 100% sure.

Reviewer: Good point, this is similar to vercel. The documentation link reminds me, does it work same way as vercel in dev mode - with a configurable local alternative?

Author: So we currently use just "sharp" for local dev, since squoosh is deprecated right?
But I don't see any big issue with adding the devImageService config flag.

Reviewer: I don't think the effort is necessary. LGTM.

> Determines which image service is used by the adapter. The following options are available, if no option is provided the adapter automaticly overrides to passtrough mode if an uncompatible image service is used, otherwise it just uses the configured image service:

Reviewer 2: ```suggestion
Determines which image service is used by the adapter. The following options are available, if no option is provided the adapter will default to  `passthrough ` mode when an incompatible image service is configured. Otherwise, it will use the globally configured image service:
```

Author: Yeah that is better than my version, but I think it is still quite hard to understand the default option, but that is not very straightforward technically.

Reviewer 2: Fair enough 😁, either way try to fix the typos in among those lines: 
- automaticly => automatically
- uncompatible => incompatible

Reviewer 2: We can wait for someone else to jump in and tweak it even more 😁

> Determines which image service is used by the adapter. The following options are available, if no option is provided the adapter will default to `passthrough ` mode when an incompatible image service is configured. Otherwise, it will use the globally configured image service:

Reviewer 3: ```suggestion
Determines which image service is used by the adapter. The adapter will default to `passthrough ` mode when an incompatible image service is configured. Otherwise, it will use the globally configured image service:
```
